Description: Update BRO_HTTP parser for Bro v2.2 
 .
 securityonion-elsa-extras (20131117-1ubuntu0securityonion4) precise; urgency=low
 .
   * Fix class nums in bro_ftp and bro_syslog
Author: Scott Allen Runnels <srunnels@gmail.com>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- securityonion-elsa-extras-20131117.orig/contrib/securityonion-patterndb.xml
+++ securityonion-elsa-extras-20131117/contrib/securityonion-patterndb.xml
@@ -899,53 +899,6 @@
 		</rules>
 	</ruleset>
 	<ruleset>
-		<pattern>bro_http</pattern>
-		<rules>
-			<rule class="19" id="19">
-				<patterns>
-					<!--ts      uid     id.orig_h       id.orig_p       id.resp_h       id.resp_p       trans_depth     method  
-					  host    uri     referrer        user_agent      request_body_len        response_body_len       status_code     
-					  status_msg      info_code       info_msg        filename        tags    username        password        
-					  proxied mime_type       md5     extraction_file-->
-					<pattern>@ESTRING::|@@ESTRING::|@@ESTRING:i0:|@@ESTRING:i1:|@@ESTRING:i2:|@@ESTRING:i3:|@@NUMBER::@|@ESTRING:s0:|@@ESTRING:s1:|@@ESTRING:s2:|@@ESTRING:s3:|@@ESTRING:s4:|@@ESTRING::|@@ESTRING:i5:|@@ESTRING:i4:|@</pattern>
-					<!--ts      uid     id.orig_h       id.orig_p       id.resp_h       id.resp_p       method  
-					host    uri     referrer        user_agent      request_body_len       
-					 request_body_interrupted        response_body_len       response_body_interrupted       status_code     
-					 status_msg      filename        tags    username        password        proxied mime_type
-       md5     extraction_file-->
-					<pattern>@ESTRING::|@@ESTRING::|@@ESTRING:i0:|@@ESTRING:i1:|@@ESTRING:i2:|@@ESTRING:i3:|@@ESTRING:s0:|@@ESTRING:s1:|@@ESTRING:s2:|@@ESTRING:s3:|@@ESTRING:s4:|@@ESTRING::|@@ESTRING::|@@ESTRING:i5:|@@ESTRING::|@@ESTRING:i4:|@</pattern>
-				</patterns>
-				<examples>
-					<example>
-						<test_message program="bro_http">1319824864.447838|g6XHk1uplZc|10.0.0.0|19427|80.175.58.76|80|GET|www.google.com|/|http://example.com|myagent|-|-|1000|0|200|</test_message>
-						<!-- srcip -->
-						<test_value name="i0">10.0.0.0</test_value>
-						<!-- srcport -->
-						<test_value name="i1">19427</test_value>
-						<!-- dstip -->
-						<test_value name="i2">80.175.58.76</test_value>
-						<!-- dstport -->
-						<test_value name="i3">80</test_value>
-						<!-- method -->
-						<test_value name="s0">GET</test_value>
-						<!-- host -->
-						<test_value name="s1">www.google.com</test_value>
-						<!-- uri -->
-						<test_value name="s2">/</test_value>
-						<!-- referer -->
-						<test_value name="s3">http://example.com</test_value>
-						<!-- user_agent -->
-						<test_value name="s4">myagent</test_value>
-						<!-- status_code -->
-						<test_value name="i4">200</test_value>
-						<!-- content_length -->
-						<test_value name="i5">1000</test_value>
-					</example>
-				</examples>
-			</rule>
-		</rules>
-	</ruleset>
-	<ruleset>
 		<pattern>bro_conn</pattern>
 		<rules>
 			<rule class="20" id="20">
--- securityonion-elsa-extras-20131117.orig/contrib/securityonion_parsers_sql.sh
+++ securityonion-elsa-extras-20131117/contrib/securityonion_parsers_sql.sh
@@ -88,3 +88,7 @@ mysql -u$MYSQL_ROOT_USER $MYSQL_PASS_SWI
 mysql -u$MYSQL_ROOT_USER $MYSQL_PASS_SWITCH $MYSQL_NODE_DB -e 'REPLACE INTO fields_classes_map (class_id, field_id, field_order) VALUES( (SELECT id FROM classes WHERE class="BRO_DHCP"), (SELECT id FROM fields WHERE field="mac"), 11);'
 mysql -u$MYSQL_ROOT_USER $MYSQL_PASS_SWITCH $MYSQL_NODE_DB -e 'REPLACE INTO fields_classes_map (class_id, field_id, field_order) VALUES( (SELECT id FROM classes WHERE class="BRO_DHCP"), (SELECT id FROM fields WHERE field="assigned_ip"), 9);'
 mysql -u$MYSQL_ROOT_USER $MYSQL_PASS_SWITCH $MYSQL_NODE_DB -e 'REPLACE INTO fields_classes_map (class_id, field_id, field_order) VALUES( (SELECT id FROM classes WHERE class="BRO_DHCP"), (SELECT id FROM fields WHERE field="lease_time"), 10);'
+# Support new BRO_HTTP format
+mysql -u$MYSQL_ROOT_USER $MYSQL_PASS_SWITCH $MYSQL_NODE_DB -e 'INSERT IGNORE INTO fields_classes_map(class_id, field_id, field_order) VALUES( (SELECT id FROM classes WHERE class="BRO_HTTP"), (SELECT id FROM fields WHERE field="mime_type"), 16);'
+mysql -u$MYSQL_ROOT_USER $MYSQL_PASS_SWITCH $MYSQL_NODE_DB -e 'REPLACE INTO fields_classes_map(class_id, field_id, field_order) VALUES( (SELECT id FROM classes WHERE class="BRO_HTTP"), (SELECT id FROM fields WHERE field="status_code"), 10);'
+mysql -u$MYSQL_ROOT_USER $MYSQL_PASS_SWITCH $MYSQL_NODE_DB -e 'REPLACE INTO fields_classes_map(class_id, field_id, field_order) VALUES( (SELECT id FROM classes WHERE class="BRO_HTTP"), (SELECT id FROM fields WHERE field="content_length"), 9);'
--- /dev/null
+++ securityonion-elsa-extras-20131117/contrib/parsers/bro_http
@@ -0,0 +1,39 @@
+<ruleset name="bro_http" id='19'>
+  <pattern>bro_http</pattern>
+  <rules>
+    <rule provider="Security_Onion" class="19" id="19">
+      <patterns>
+        <pattern>@ESTRING::|@@ESTRING::|@@ESTRING:i0:|@@ESTRING:i1:|@@ESTRING:i2:|@@ESTRING:i3:|@@ESTRING::|@@ESTRING:s0:|@@ESTRING:s1:|@@ESTRING:s2:|@@ESTRING:s3:|@@ESTRING:s4:|@@ESTRING::|@@ESTRING:i4:|@@ESTRING:i5:|@@ESTRING::|@@ESTRING::|@@ESTRING::|@@ESTRING::|@@ESTRING::|@@ESTRING::|@@ESTRING::|@@ESTRING::|@@ESTRING::|@@ESTRING::|@@ESTRING:s5:|@</pattern>
+      </patterns>
+      <examples>
+        <example>
+          <test_message program="bro_http">1384366977.167889|CeEWn14oWSULhfVyL6|192.168.4.226|58543|74.125.228.32|80|1|GET|clients2.google.com||-|Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/28.0.1500.71 Chrome/28.0.1500.71 Safari/537.36|0|380|200|OK|-|-|-|(empty)|-|-|-|-|-|F6XqFl1L5YkUsflHK8|application/xml</test_message>
+          <!-- srcip -->
+          <test_value name="i0">192.168.4.226</test_value>
+          <!-- srcport -->
+          <test_value name="i1">58543</test_value>
+          <!-- dstip -->
+          <test_value name="i2">74.125.228.32</test_value>
+          <!-- dstport -->
+          <test_value name="i3">80</test_value>
+          <!-- method -->
+          <test_value name="s0">GET</test_value>
+          <!-- host -->
+          <test_value name="s1">clients2.google.com</test_value>
+          <!-- uri -->
+          <test_value name="s2">/service/update2/crx?os=linux&arch=x64&nacl_arch=x86-64&prod=chrome&prodchannel=unknown&prodversion=28.0.1500.71&x=id=npdjjkjlcidkjlamlmmdelcjbcpdjocm&v=0.0.0.0&uc&x=id=hfnkpimlhhgieaddgfemjhofmfblmnib&v=1301&uc|-|Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/28.0.1500.71 Chrome/28.0.1500.71 Safari/537.36|0|380|200|OK|-|-|-|(empty)|-|-|-|-|-|F6XqFl1L5YkUsflHK8|application/xml</test_message></test_value>
+          <!-- referer -->
+          <test_value name="s3">-</test_value>
+          <!-- user_agent -->
+          <test_value name="s4">Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/28.0.1500.71 Chrome/28.0.1500.71 Safari/537.36</test_value>
+          <!-- resp_body_len -->
+          <test_value name="i4">380</test_value>
+          <!-- status_code -->
+          <test_value name="i5">200</test_value>
+          <!-- mime_type -->
+          <test_value name="s5">-</test_value>
+        </example>
+      </examples>
+    </rule>
+  </rules>
+</ruleset>
